ext/TerrariumCheckpointingExt/TerrariumCheckpointingExt.jl

```diff
@@ -0,0 +1,33 @@
+module TerrariumCheckpointingExt
+
+using DocStringExtensions
+using Checkpointing
+
+using Terrarium
+using Terrarium: run!
+
+"""
+    $SIGNATURES
+
+Run the simulation for `steps` with timestep size `Δt` (in seconds or Dates.Period).
+
+Uses a `checkpointing_scheme` from Checkpointing.jl to checkpoint the integrator state for automatic differentiation with Enzyme.jl. 
+"""
+function Terrarium.run!(
+        integrator::ModelIntegrator,
+        checkpointing_scheme::Checkpointing.Scheme,
+        steps::Int,
+        Δt = default_dt(timestepper(integrator))
+    )
+    Δt = Terrarium.convert_dt(Δt)
+
+    @ad_checkpoint checkpointing_scheme for _ in 1:steps
```

Comment:
Could you briefly explain what this actually does?

Reply:
The macro applies a checkpointing scheme to the loop it is marking. So intermediate values in the (forward pass of the) AD are only saved at those points. How often a checkpoint is saved depends on the scheme and the setting, both is set in the script. Here I use the `Revolve` scheme (https://dl.acm.org/doi/10.1145/347837.347846) and save the checkpoint at every time step. In this Julia implementation the inner body has to be differentiated by Enzyme, but for the outer body even other AD systems would work.

Reply:
There's a really great side-effect of Checkpointing for Enzyme: it massively reduces compile time. 

Without Checkpointing Enzyme would look at building and compiling the computational graph of the complete iteration over all time steps. With Checkpointing Enzyme just differentiates the body of the time loop. It just has to compile this gradient once, and can then recompute intermediate values very efficiently.